src/brsTypes/components/RoSGNode.ts

```diff
@@ -1914,3 +1925,55 @@ function addChildren(
         }
     });
 }
+
+function cloneNode(
+    interpreter: Interpreter,
+    toBeCloned: RoSGNode,
+    isDeepCopy: Boolean
+): RoSGNode | BrsInvalid {
+    let copy = createNodeByType(interpreter, new BrsString(toBeCloned.nodeSubtype));
+
+    let originalFields = toBeCloned.getFields();
+    let copiedFields = new RoAssociativeArray([]);
+    let addReplace = copiedFields.getMethod("addreplace");
+
+    // Copy the fields.
+    for (let [key, value] of originalFields) {
+        if (addReplace) {
+            addReplace.call(
+                interpreter,
+                new BrsString(key),
+                getBrsValueFromFieldType(value.getType(), value.toString())
+            );
+        }
+    }
```

Comment:
All this can be moved into the `if copy` since it won't be used if copy isn't valid.